optking/stepAlgorithms.py

```diff
@@ -1,825 +1,803 @@
+        """Check whether eigenvector of RFO matrix is numerically acceptable for following and of
+        proper symmetry. Double checks max values #TODO add real symmetry check
+
+        Parameters
+        ----------
+        vector: np.ndarray
+            an eigenvector of the rfo matrix
+        index: int
+            sorted index for the vector (eigenvector)
+        """
+
+        def reject_root(mesg):
+            self.logger = logging.getLogger(__name__)
+            self.logger.warning("\tRejecting RFO root %d because %s", index + 1, mesg)
+            return False
+
+        self.logger.info(print_array_string(vector))
+        # Check symmetry of root. Leave True if unessecary. Not currently functioning
+        symmetric = True if not self.accept_symmetry_breaking else is_dq_symmetric(self.molsys, vector[:-1])
+
+        if not symmetric:
+            return reject_root("it breaks the molecular point group")
+
+        if vector[-1] < 1e-10:
+            return reject_root("Normalization gives large value")
+
+        if np.amax(np.abs(vector)) > self.rfo_normalization_max:
+            return reject_root("Normalization gives large value")
+
+        return True
 
-        else:
-            logger.debug("\tSearching with smaller step between first 2 points.")
-            s *= 0.5
-            Ec = Eb
-            Eb = 0
 
-    # get norm |q| and unit vector in the step direction
-    ls_dqnorm = np.linalg.norm(dq_achieved)
-    ls_u = dq_achieved / ls_dqnorm
+class PartitionedRFO(RFO):
+    """Partitions the gradient augmented Hessian into eigenvectors to maximize along (direction of the TS)
+    and minimize along all other directions. Rational Function or (2x2 Pade approximation)"""
 
-    # get gradient and hessian in step direction
-    ls_g = -1 * np.dot(fq, ls_u)  # should be unchanged
-    ls_h = np.dot(ls_u, np.dot(H, ls_u))
+    def step(self, fq, H, *args, **kwargs):
 
-    if op.Params.print_lvl > 1:
-        logger.info("\n\\t|target step|: %15.10f" % ls_dqnorm)
-        logger.info("\tLS_gradient     : %15.10f" % ls_g)
-        logger.info("\tLS_hessian      : %15.10f" % ls_h)
+        hdim = len(fq)  # size of Hessian
 
-    DEprojected = de_projected("NR", ls_dqnorm, ls_g, ls_h)
-    logger.debug("\tProjected quadratic energy change using full Hessian: %15.10f\n" % DEprojected)
+        # Diagonalize H (technically only have to semi-diagonalize)
+        h_eig_values, h_eig_vectors = symm_mat_eig(H)
+        hess_diag = np.diag(h_eig_values)
 
-    oHistory.nuclear_repulsion_energy = computer.trajectory[-1]["properties"]["nuclear_repulsion_energy"]
+        if self.print_lvl > 2:
+            self.logger.info("\tEigenvalues of Hessian\n\n\t%s", print_array_string(h_eig_values))
+            self.logger.info("\tEigenvectors of Hessian (rows)\n%s", print_mat_string(h_eig_vectors))
+            self.logger.debug("\tFor P-RFO, assuming rfo_root=1, maximizing along lowest eigenvalue of Hessian.")
```

Comment:
I thought the earlier code was written consistent with the search for higher-order saddle points.  (though I don't recall anyone ever did that).  Maybe that was the C++ code only?  Same algorithm.

Reply:
I thought the code worked with mu > 1.